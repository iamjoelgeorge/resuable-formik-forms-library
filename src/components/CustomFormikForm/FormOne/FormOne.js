import React from 'react';

import styles from './FormOne.module.scss';
import FormContainer from '../FormContainer/FormContainer';
import FormBody from './FormBody/FormBody';

const FormOne = () => {
  const initialValues = {
    name: '',
    email: '',
    age: '',
    test: '',
    description: '',
    license: false,
    idProofs: [],
    formOneRadioGroup: 'two',
    departureDate: new Date(),
    returnDate: new Date(),
  };

  const validations = [
    {
      name: 'name',
      type: 'string',
      isRequired: false,
      message: 'Enter your name',
      minChars: { num: 2, message: 'Please enter at least 2 characters' },
    },
    // {
    //   name: 'email',
    //   type: 'email',
    //   isRequired: true,
    //   message: 'Enter the correct email id',
    // },
    // {
    //   name: 'description',
    //   type: 'string',
    //   isRequired: false,
    //   message: 'Please add a comment.',
    //   minChars: { num: 2, message: 'Please enter at least 2 characters' },
    //   maxChars: { num: 5, message: 'You can only enter upto 5 characters' },
    // },
    // {
    //   name: 'age',
    //   type: 'number',
    //   isRequired: true,
    //   message: 'Tell us your age',
    // },
    // {
    //   name: 'idProofs',
    //   type: 'file',
    //   isRequired: true,
    //   message: 'You can only upload png images.',
    // },
  ];

  const handleSubmit = (values) => console.log('Form 1 submitted:', values);

  return (
    <FormContainer
      initialValues={initialValues}
      validations={validations}
      containerClass={styles.container}
      onSubmit={handleSubmit}
    >
      <FormBody />
    </FormContainer>
  );
};

export default FormOne;
