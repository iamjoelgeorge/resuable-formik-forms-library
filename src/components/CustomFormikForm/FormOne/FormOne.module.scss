@import '../../../theme/variables.scss';

.container {
  display: flex;
  flex-direction: column;
  background-color: $white;
  border: 1px solid $border;
  padding: 20px;
  padding-top: 0;
  width: 350px;
}
