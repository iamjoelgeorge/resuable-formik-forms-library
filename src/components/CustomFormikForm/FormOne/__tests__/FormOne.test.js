import React from 'react';
import { cleanup, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import FormOne from '../FormOne';

afterEach(cleanup);
describe('[Component]: FormOne', () => {
  it('should match the Snapshot', () => {
    const tree = renderer.create(<FormOne />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render and submit the form', async () => {
    // const handleSubmit = jest.fn();
    render(<FormOne />);
    // userEvent.type(screen.getByLabelText(/name/i), 'John');

    // await waitFor(() =>
    //   expect(handleSubmit).toHaveBeenCalledWith({
    //     name: 'John',
    //   }),
    // );
  });
});
