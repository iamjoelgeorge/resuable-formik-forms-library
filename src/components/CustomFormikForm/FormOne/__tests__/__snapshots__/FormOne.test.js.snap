// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: FormOne should match the Snapshot 1`] = `
<form
  action="#"
  className="container"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <h1>
    Form 1
  </h1>
  <div
    className="container "
  >
    <div
      className="inputWithTooltipContainer"
    >
      <input
        className={
          Array [
            "input",
          ]
        }
        data-testid="input-test"
        disabled={true}
        id="test"
        name="test"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        type="text"
        value=""
      />
      <div
        className="container  "
      >
        <label
          htmlFor="test"
        >
          Disabled Input
        </label>
      </div>
    </div>
    
    
    
  </div>
  <div>
    <div
      className="testContainer"
    >
      <div
        className="container input"
      >
        <div
          className="inputWithTooltipContainer"
        >
          <input
            className={
              Array [
                "input",
              ]
            }
            data-testid="name"
            disabled={false}
            id="name"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            type="text"
            value=""
          />
          <div
            className="container  "
          >
            <label
              htmlFor="name"
            >
              Name
            </label>
          </div>
        </div>
        
        
        
      </div>
      <div
        className="container input"
      >
        <div
          className="inputWithTooltipContainer"
        >
          <input
            className={
              Array [
                "input",
              ]
            }
            data-testid="input-age"
            disabled={false}
            id="age"
            name="age"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Tell us how old you are"
            type="text"
            value=""
          />
          <div
            className="container   slideLabel"
          >
            <label
              htmlFor="age"
            >
              Age (with placeholder)
            </label>
          </div>
        </div>
        
        
        <div
          className="common tooltipContainer "
        >
          <a
            href="#"
            rel="noreferrer"
            target="_blank"
          >
            This is a Tooltip link?
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container "
  >
    <div
      className="inputWithTooltipContainer"
    >
      <input
        className={
          Array [
            "input",
          ]
        }
        data-testid="input-email"
        disabled={false}
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        type="email"
        value=""
      />
      <div
        className="container  "
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
      </div>
      <div
        className="toolTipWithImage"
      >
        <div
          className="container "
        >
          <img
            alt="tooltip"
            onClick={[Function]}
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
    
    <div
      className="common "
    >
      <a
        href="#"
        rel="noreferrer"
        target="_blank"
      >
        This is a Help link?
      </a>
    </div>
    
  </div>
  <div>
    <div
      className="container"
      id="custom-calendar-date-picker"
    >
      <div>
        <button
          className="selectedDateContainer"
          onClick={[Function]}
          type="button"
        >
          <div
            className="container   slideLabel"
          >
            <label
              htmlFor="selectedDate"
            >
              Departure Date
            </label>
          </div>
          <p
            className="selectedDate"
            id="selectedDate"
          >
            Fri, 17 Sep 2021
            <span
              className="dropdownIcon"
            >
              <img
                alt="Dropdown icon"
                src="test-file-stub"
              />
            </span>
          </p>
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      className="container"
    >
      <p
        className="label"
      >
        Return Date
      </p>
      <div
        className="dateContainer"
      >
        <div
          className="date"
        >
          <div
            className="container"
            onClick={[Function]}
          >
            <p
              className="selectedDate"
            >
              
              <span
                className="dropdownIconContainer"
              >
                <img
                  alt="Dropdown icon"
                  src="test-file-stub"
                />
              </span>
            </p>
          </div>
        </div>
        <div
          className="month"
        >
          <div
            className="container"
            onClick={[Function]}
          >
            <p
              className="selectedDate"
            >
              
              <span
                className="dropdownIconContainer"
              >
                <img
                  alt="Dropdown icon"
                  src="test-file-stub"
                />
              </span>
            </p>
          </div>
        </div>
        <div
          className="year"
        >
          <div
            className="container"
            onClick={[Function]}
          >
            <p
              className="selectedDate"
            >
              
              <span
                className="dropdownIconContainer"
              >
                <img
                  alt="Dropdown icon"
                  src="test-file-stub"
                />
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container "
  >
    <div
      className="inputWithTooltipContainer"
    >
      <input
        className={
          Array [
            "input",
          ]
        }
        data-testid="input-test"
        disabled={false}
        id="test"
        name="test"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        type="number"
        value=""
      />
      <div
        className="container  "
      >
        <label
          htmlFor="test"
        >
          Test Input
        </label>
      </div>
      <div
        className="toolTipWithImage"
      >
        <div
          className="container "
        >
          <img
            alt="tooltip"
            onClick={[Function]}
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
    <div
      className="common "
    >
      <p
        className="optionalText"
      >
        Optional
      </p>
    </div>
    
    
  </div>
  <div
    className="container description"
  >
    <div
      className="fieldContainer"
    >
      <textarea
        className="textarea"
        data-testid="textarea-description"
        disabled={false}
        name="description"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        value=""
      />
      <div
        className="container  label"
      >
        <label
          htmlFor="description"
        >
          Additional Details
        </label>
        <div
          className="container tooltip"
        >
          <img
            alt="tooltip"
            onClick={[Function]}
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
    <div
      className="common "
    >
      <p
        className="optionalText"
      >
        This is your textarea optional text
      </p>
    </div>
    
    
  </div>
  <div
    className="container"
  >
    <div
      className="container  componentHeading slideLabel"
    >
      <label
        htmlFor="license"
      >
        License
      </label>
    </div>
    <div
      className="optionWithTooltip"
    >
      <label
        className="optionLabel"
      >
        <input
          checked={false}
          className="checkboxInput"
          data-testid="checkbox-license"
          disabled={false}
          name="license"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
          value={false}
        />
        <span
          className="presentationalCheckbox"
        >
          <span
            className="checkboxChecked"
          />
        </span>
        <span
          className="labelSpanElement"
        >
          Do you have a license?
        </span>
      </label>
      <div
        className="container tooltip"
      >
        <img
          alt="tooltip"
          onClick={[Function]}
          src="test-file-stub"
        />
      </div>
    </div>
    
    <div
      className="common optionalText"
    >
      <a
        href="#"
        rel="noreferrer"
        target="_blank"
      >
        This link is for your help
      </a>
    </div>
    
  </div>
  <div
    className="container"
  >
    <div
      className="container  componentHeading slideLabel"
    >
      <label
        htmlFor="formOneRadioGroup"
      >
        Radio Group
      </label>
      <div
        className="container tooltip"
      >
        <img
          alt="tooltip"
          onClick={[Function]}
          src="test-file-stub"
        />
      </div>
    </div>
    <div
      aria-labelledby="formOneRadioGroup"
      className="radioGroup"
      role="group"
    >
      <div
        className="radioButtonWithLabel"
      >
        <label
          className="optionLabel"
        >
          <input
            checked={false}
            className="radioInput"
            disabled={false}
            name="formOneRadioGroup"
            onBlur={[Function]}
            onChange={[Function]}
            type="radio"
            value="one"
          />
          <span
            className="presentationalRadioButton"
          >
            <span
              className="radioButtonSelected"
            />
          </span>
          <span
            className="labelSpanElement"
          >
            One Radio
          </span>
        </label>
      </div>
      <div
        className="radioButtonWithLabel"
      >
        <label
          className="optionLabel"
        >
          <input
            checked={true}
            className="radioInput"
            disabled={false}
            name="formOneRadioGroup"
            onBlur={[Function]}
            onChange={[Function]}
            type="radio"
            value="two"
          />
          <span
            className="presentationalRadioButton"
          >
            <span
              className="radioButtonSelected"
            />
          </span>
          <span
            className="labelSpanElement"
          >
            Two Radio
          </span>
        </label>
      </div>
      <div
        className="radioButtonWithLabel"
      >
        <label
          className="optionLabel"
        >
          <input
            checked={false}
            className="radioInput"
            disabled={false}
            name="formOneRadioGroup"
            onBlur={[Function]}
            onChange={[Function]}
            type="radio"
            value="three"
          />
          <span
            className="presentationalRadioButton"
          >
            <span
              className="radioButtonSelected"
            />
          </span>
          <span
            className="labelSpanElement"
          >
            Three Radio
          </span>
        </label>
        <div
          className="container tooltip"
        >
          <img
            alt="tooltip"
            onClick={[Function]}
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
    
    
    <div
      className="common tooltipContainer optionalText"
    >
      <a
        href="https://www.google.com"
        rel="noreferrer"
        target="_blank"
      >
        Visit Google to know more.
      </a>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="container  label slideLabel"
    >
      <label
        htmlFor="idProofs"
      >
        Upload your documents
      </label>
      <div
        className="container tooltip"
      >
        <img
          alt="tooltip"
          onClick={[Function]}
          src="test-file-stub"
        />
      </div>
    </div>
    <div>
      <input
        hidden="hidden"
        multiple={true}
        onChange={[Function]}
        type="file"
      />
      <button
        className="buttonToOpenFileUploader"
        onClick={[Function]}
        type="button"
      >
        Upload files
      </button>
    </div>
    <div
      className="common optionalText"
    >
      <p
        className="optionalText"
      >
        test
      </p>
    </div>
    
    
  </div>
  <button
    className="button purpleTheme "
    data-testid="button-Submit"
    disabled={false}
    type="submit"
  >
    Submit
  </button>
</form>
`;
